plSqlStandards.md

```diff
@@ -0,0 +1,316 @@
+﻿# PL/SQL Coding Standards
+
+## Rationale
+
+This document is intended to present the standards to which all PL/SQL
+code produced by, or on behalf of, DEFRA should adhere. PL/SQL coding
+standards are intended to:
+
+  - Make the produced code more easily read and understood.
+
+  - Make the code more easily maintainable.
+
+  - Make the code more reliable.
+
+## The Derivation of these Standards
+
+These standards have been derived from hands-on experience at the
+Environment Agency and Rural Payments Agency over many years. As such
+they are mature, comprehensive while also being reasonably lightweight
+and follow standard industry practice.
+
+## Important Recommendation: Refer to these Standards by Version as well as Name.
+
+These standards are likely to be updated on an ongoing basis. Anything
+using these standards should therefore refer to these standards not only
+be name (“PL/SQL Coding Standards”) but also by version (“1.0”). It
+should not be assumed that established products will be continually
+retrospectively updated to always reflect the very latest iteration of
+these standards: therefore recording which specific version of these
+standards has been adopted (and keeping all the preceding versions of
+these standards) is essential.
+
+# General Conventions
+
+## Structure
+
+### Use Block and Label coding for loops and anonymous blocks within procedures.
+
+For example:
+
+> CREATE OR REPLACE PROCEDURE my\_proc
```

Comment:
All of this (and the other bits of code in here) should be formatted - I would suggest using `code` pre-formatting:

```
CREATE OR REPLACE PROCEDURE my_proc
IS
BEGIN
   <<Block1>>
DECLARE
   ......

```
etc.

Reply:
You can actually go beyond just code formatting.

```pgsql
CREATE OR REPLACE PROCEDURE my_proc
IS
BEGIN
   <<Block1>>
DECLARE
   ......
```

GItHub uses [linguist](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml) to provide syntax highlighting so anything listed there can be specified on your code block.

N.B. Though it appears this code block doesn't give the best example of the syntax highlighting 😁